### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: src/_locales/en_US/messages.json [hunk 1]
```diff
@@ -135,6 +147,15 @@
...
     "message": "Terms & Conditions"
   },
+  "viewBetaDecision": {
+    "message": "TBD"
+  },
+  "viewBetaDecisionButton": {
+    "message": "Learn more"
+  },
+  "viewBetaDecisionAlreadyMade": {
+    "message": "TBD decision made?"
```
Comment (jonathanKingston) [inline src/_locales/en_US/messages.json]: Perhaps put some real placeholder text in these in case we want to merge?

File: src/commons/utils.js [hunk 2]
```diff
@@ -18,6 +18,9 @@ const PROXY_STATE_INACTIVE = "inactive";
 // The user is registered, the proxy is active.
 const PROXY_STATE_ACTIVE = "active";
 
+// The beta user has to made a decision.
```
Comment (jonathanKingston) [inline src/commons/utils.js]: s/made/make/

File: src/background/main.js [hunk 3]
```diff
@@ -386,6 +394,27 @@ class Main {
...
+    // Compute proxy state does the magic.
+    if (await this.computeProxyState()) {
+      await this.ui.update();
+    }
+  }
+
+  async betaDecisionMade() {
+    await this.fxa.refreshProfileData();
+
+    // Compute proxy state does the magic.
```
Comment (jonathanKingston) [inline src/background/main.js]: Make a common helper function between this and the contents of `betaDecisionRequired`?

File: src/background/migrationData.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        return false;
+      }
+    }
+
+    // eslint-disable-next-line verify-await/check
+    let now = Date.now();
+    let nowInSecs = Math.round(now / 1000);
+
+    this.migrationReceivedAt = nowInSecs;
+    ConfigUtils.migrationData = json;
```
Comment (jonathanKingston) [inline src/background/migrationData.js]: This seems like a code smell, why are we even using this via the global and not `this.migrationData`?

**Comment by jonathanKingston** [comment]
This looks good to me.

A nice to have as we discussed is refresh the user profile as a blocking load as part of opening the doorhanger every time once the time limit is up.

Also we are blocking this on the text and the other TODO items within.

Are there any other changes we need?

File: src/_locales/en_US/messages.json (hunk 1 above)
Comment (bakulf) [reply src/_locales/en_US/messages.json]: We don't have this yet.

**Comment by bakulf** [comment]
> A nice to have as we discussed is refresh the user profile as a blocking load as part of opening the doorhanger every time once the time limit is up.

Follow up. I open a separate issue for this.

> Also we are blocking this on the text and the other TODO items within.

Well.. better to have TODOs than random code in place :)

> Are there any other changes we need?

No. the limit will be implemented in a separate patch.

File: src/background/migration.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
+import {Component} from "./component.js";
+import {migrationData} from "./migrationData.js";
+import {Tier} from "./tier.js";
+
+const TIER_ENDPOINT_URLS = [
+ // [ URL, redirect to ]
+ [ "https://private-network-beta-landing-test.stage.mozaws.net/free-tier-enabled-install", "https://private-network-beta-landing-test.stage.mozaws.net/free-tier-enabled" ],
```
Comment (pdehaan) [inline src/background/migration.js]: Some of these are dev URLs and some are prod (which I _assume_ is on purpose), but... possibly worth noting is that the staging URL was meant to be internal and never public. /oops

File: src/background/migrationData.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    return await this.forceMigrationData(json);
+  }
+
+  async forceMigrationData(json) {
+    const fields = [ "expirationTime" ];
+    for (let field in fields) {
+      if (!(fields[field] in json)) {
+        console.log(`Field ${fields[field]} required in the migration json!`);
```
Comment (pdehaan) [inline src/background/migrationData.js]: Should this be using conditional logging via `log()` function so we only log while in `debugingMode`?

Comment (bakulf) [reply src/background/migrationData.js]: console.error() seems better here.

**Pull request closed without merging**